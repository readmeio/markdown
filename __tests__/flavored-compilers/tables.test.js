import { mdast, md } from '../../index';

describe('table compiler', () => {
  it('converts to markdown syntax', () => {
    const markdown = `
[block:parameters]
{
  "data": {
    "h-0": "th 1",
    "h-1": "th 2",
    "0-0": "cell 1",
    "0-1": "cell 2"
  },
  "cols": 2,
  "rows": 1,
  "align": [
    "center",
    "center"
  ]
}
[/block]
`;

    expect(md(mdast(markdown))).toBe(
      `|  th 1  |  th 2  |
| :----: | :----: |
| cell 1 | cell 2 |
`
    );
  });

  it('saves to magic block syntax if there are breaks', () => {
    const markdown = `
[block:parameters]
{
  "data": {
    "h-0": "th 1",
    "h-1": "th 2",
    "0-0": "cell 1\\nextra line",
    "0-1": "cell 2"
  },
  "cols": 2,
  "rows": 1,
  "align": [
    "center",
    "center"
  ]
}
[/block]
`;

    expect(md(mdast(markdown))).toBe(`[block:parameters]
{
  "data": {
    "h-0": "th 1",
    "h-1": "th 2",
    "0-0": "cell 1  \\nextra line",
    "0-1": "cell 2"
  },
  "cols": 2,
  "rows": 1,
  "align": [
    "center",
    "center"
  ]
}
[/block]
`);
  });

  it('converts to magic block syntax if there are breaks', () => {
    const markdown = `
|  th 1  |  th 2  |
| :----: | :----: |
| cell 1 | cell 2 |
`;
    const nodes = mdast(markdown);
    const cell = nodes.children[0].children[1].children[0];
    cell.children = [...cell.children, { type: 'break' }, { type: 'text', value: 'extra line' }];

    expect(md(nodes)).toBe(`[block:parameters]
{
  "data": {
    "h-0": "th 1",
    "h-1": "th 2",
    "0-0": "cell 1  \\nextra line",
    "0-1": "cell 2"
  },
  "cols": 2,
  "rows": 1,
  "align": [
    "center",
    "center"
  ]
}
[/block]
`);
  });
});
