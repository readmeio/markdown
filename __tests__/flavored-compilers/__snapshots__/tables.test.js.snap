// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table compiler converts to magic block syntax if there are breaks 1`] = `
"
[block:parameters]{\\"data\\":{\\"h-0\\":\\"th 1\\",\\"h-1\\":\\"th 2\\",\\"0-0\\":\\"cell 1  \\\\nextra line\\",\\"0-1\\":\\"cell 2\\"},\\"cols\\":2,\\"rows\\":1,\\"align\\":[\\"center\\",\\"center\\"]}[/block]
"
`;

exports[`table compiler saves to magic block syntax if there are breaks 1`] = `
"
[block:parameters]{\\"data\\":{\\"h-0\\":\\"th 1\\",\\"h-1\\":\\"th 2\\",\\"0-0\\":\\"cell 1  \\\\nextra line\\",\\"0-1\\":\\"cell 2\\"},\\"cols\\":2,\\"rows\\":1,\\"align\\":[\\"center\\",\\"center\\"]}[/block]
"
`;
